jl4-lsp/src/LSP/SemanticTokens.hs

```diff
@@ -34,7 +34,7 @@ runSemanticTokensM semTokenCtx a = Except.runExcept $ ReaderT.runReaderT (toSemT
 --
 -- We might want to override some functionality here. We should perhaps
 -- try to find another way to do this.
-class ToSemTokens c t a | a t -> c where
+class ToSemTokens c t a | a c -> t where
```

Comment:
is this a bug fix? Or how is this change in semantics justified? Does this actually encode a rule for the type class or is this just for slightly improving type inference when inferring the instances?

Reply:
This is only for type inference, looks like we don't need it at all.